react/src/lib/components/DeckGLMap/DeckGLMap.stories.tsx

```diff
@@ -340,3 +346,222 @@ DepthTest.parameters = {
+    "@@type": "MapLayer",
+    id: "mesh-layer",
+    meshUrl: "hugin_depth_25_m.float32",
+    frame: {
+        origin: [432150, 6475800],
+        count: [291, 229],
+        increment: [25, 25],
+        rotDeg: 0,
+    },
+    propertiesUrl: "kh_netmap_25_m.float32",
+    contours: [0, 100],
+    isContoursDepth: true,
+    gridLines: false,
+    material: false,
+    colorMapName: "Physics",
+};
+
+//eslint-disable-next-line
+const MultiColorSelectorTemplate = (args: any) => {
+    const [getColorName, setColorName] = React.useState("Stratigraphy");
+    const [isLog, setIsLog] = React.useState(false);
+    const wellLayerData = React.useCallback((data) => {
+        setColorName(data.name ? data.name : data.legendColorName);
+    }, []);
+
+    // interpolation method
+    const getInterpolateMethod = React.useCallback((data) => {
+        setIsLog(data.isLog);
+    }, []);
+
+    const [colorName, setColorName1] = React.useState("GasWater");
+    const [colorRange, setRange] = React.useState();
+    const [isAuto, setAuto] = React.useState();
+    const [breakPoints, setBreakPoint] = React.useState();
+    const [isMapLayerLog, setIsMapLayerLog] = React.useState(false);
+    const [isNearest, setIsNearest] = React.useState(false);
+
+    // user defined breakpoint(domain)
+    const userDefinedBreakPoint = React.useCallback((data) => {
+        if (data) setBreakPoint(data.colorArray);
+    }, []);
+
+    // Get color name from color selector
+    const colorNameFromSelector = React.useCallback((data) => {
+        setColorName1(data);
+    }, []);
+
+    // user defined range
+    const userDefinedRange = React.useCallback((data) => {
+        if (data.range) setRange(data.range);
+        setAuto(data.isAuto);
+    }, []);
+
+    // Get interpolation method from color selector to layer
+    const getMapLayerInterpolateMethod = React.useCallback((data) => {
+        setIsMapLayerLog(data.isLog);
+        setIsNearest(data.isNearest);
+    }, []);
+
+    // color map function
+    const colorMapFunc = React.useCallback(() => {
+        return createColorMapFunction(
+            colorName,
+            isMapLayerLog,
+            isNearest,
+            breakPoints
+        );
+    }, [colorName, isMapLayerLog, isNearest, breakPoints]);
+
+    const min = 100;
+    const max = 1000;
+
+    const layers = [
+        {
+            ...args.wellLayers[0],
+            colorMappingFunction: createColorMapFunction(getColorName),
+            logColor: getColorName ? getColorName : wellLayers[0].logColor,
+            isLog: isLog,
+        },
+        {
+            ...meshMapLayerFloat32,
+            colorMapName: colorName,
+            colorMapRange:
+                colorRange && isAuto == false ? colorRange : [min, max],
+            colorMapFunction: colorMapFunc(),
+        },
+    ];
+    return (
+        <>
+            <div className={useStyles().colorSelector}>
+                <div style={{ marginTop: 50, height: 70 }}>
```

Comment:
Could we try using the `cssLegendStyles` prop on the ColorLegend instead of nesting in a div?

Reply:
Had a try on using the View component but unluckily it still does not work. I guess it is because View component can only take one component and if it becomes two, there is some "outside" div has been created by react as I can still see this     
position: relative;
    height: 92vh;
    width: 97vw;

even if I am using the View component as suggested.